docs/specs/app-initiated-sso-to-web.md

```diff
@@ -0,0 +1,179 @@
+# App Initiated SSO to Web
+
+- [Abstract](#abstract)
+- [The Flow](#the-flow)
+- [Native App Authorization Extensions](#native-app-authorization-extensions)
+- [Token Exchange Profile for App Initiated SSO to Web](#token-exchange-profile-for-app-initiated-sso-to-web)
+  - [Defined token type identifiers](#defined-token-type-identifiers)
+  - [Token Exchange Request](#token-exchange-request)
+  - [Token Exchange Response](#token-exchange-response)
+- [Exchanging for browser cookie of a session with the authorization endpoint](#exchanging-for-browser-cookie-of-a-session-with-the-authorization-endpoint)
+- [Security Considerations](#security-considerations)
+  - [Binding Tokens To Device](#binding-tokens-to-device)
+- [Changes In SDKs](#changes-in-sdks)
+- [Usecases](#usecases)
+
+## Abstract
+
+This document describes a mechanism that allows a cookie based web app to share the authentication obtained by a mobile app.
+
+This document is drafted based on [OpenID Connect Native SSO for Mobile Apps 1.0](https://openid.net/specs/openid-connect-native-sso-1_0.html).
+
+## The Flow
+
+```mermaid
+sequenceDiagram
+    participant app1 as Mobile App #35;1
+    participant Web as Web App #35;2 <br/>(Cookie Based)<br/>https://www.example.com
+    participant Authgear as Authgear <br/> https://auth.example.com
+    app1->>+Authgear: [1]GET https://auth.example.com/oauth2/authorize<br/>client_id=1<br/>scope=x_device_browser_sso<br/>redirect_uri=app://redirect
```

Comment:
Maybe we should rename the scope to `x_device_browser_session_token`. If `x_device_browser_session_token` is granted in the scope, then device browser session token can be exchanged in the token endpoint with Token Exchange mechanism.